source/code/go/src/plugins/oms.go

```diff
@@ -253,6 +246,10 @@ func PostDataHelper(tailPluginRecords []map[interface{}]interface{}) int {
 			DataItems: dataItems}
 
 		marshalled, err := json.Marshal(logEntry)
+		if err != nil {
+			Log("Error while Marshalling log Entry: %s", err.Error())
+			return output.FLB_RETRY
```

Comment:
We should do FLB_OK (and log a metric for jsonmarshaiing error) and continue here as it could be poisonous data and there is no point in asking FB to retry as we will hit the same again..